posts/batchcorrection/index.qmd

```diff
@@ -0,0 +1,736 @@
+
+
+Here we can re-make the UMAP and cluster without batch correction for sake of comparison.
+```{r}
+#| eval: false
+#| echo: true
+
+##### Run helper function to make a umap and return the nearest-neighbors graph.
+pcaobj <- sem_comb[["pca"]]
+class(pcaobj)
+umapobj <- make_umap(list(reduction.data = pcaobj))
+
+
+```
+
+```{r}
+#| eval: false
+#| echo: false
+
+#saveRDS(umapobj, "umapobj.combined.regular.rds")
+umapobj <- readRDS("figures/umapobj.combined.regular.rds")
+
+```
+
+```{r}
+#| eval: false
+#| echo: true
+
+sem_comb[["umap"]] <- umapobj$ump
+sem_comb[["umapgrph"]] <- Seurat::as.Graph(umapobj[["grph"]])
+
+### Unsupervised clustering
+sem_comb <- Seurat::FindClusters(sem_comb, graph.name = "umapgrph")
+sem_comb@meta.data$clusters_unsup_unintegrated <- sem_comb@meta.data$seurat_clusters
+```
+
+```{r}
+#| eval: false
+#| echo: false
+
+fwrite(data.table(sem_comb@meta.data)[,.(clusters_unsup_noncorrect, cell_ID)]
+       ,file="clustersunsup.combined.regular.csv")
+
+```
+
+
+```{r}
+#| eval: false
+#| echo: false
+
+regular_clusters <- fread("figures/clustersunsup.combined.regular.csv")
+sem_comb <- 
+Seurat::AddMetaData(sem_comb
+                    ,data.frame(regular_clusters[,.(clusters_unsup_unintegrated = as.character(clusters_unsup_unintegrated))]
+                               ,row.names = regular_clusters$cell_ID)
+                    )
+
+```
+
+
+# Comparison with and without integration {#pcplotbatch}
+
+Plotting the first two PC's, colored by `batchid`, it looks like the v1.5 data may span a more similar domain of PC1 and PC2 space after Harmony, indicating better integration.
+
+```{r}
+#| eval: false
+#| code-fold: true
+
+all.equal(rownames(harmony_embeddings), sem_comb@meta.data$cell_ID)
+
+pc1vs2_harmony <- 
+cbind(data.table(sem_comb@meta.data)[,.(cell_ID, batchid, x_slide_mm, y_slide_mm)]
+      ,data.table(harmony_embeddings))
+
+harmony_plt <- 
+ggplot(pc1vs2_harmony, aes(x = PC_1, y = PC_2, color=batchid)) + geom_point(size=0.2, alpha=0.5, shape=16) + 
+  scale_color_manual(values = c("purple","darkgreen")) + 
+  theme_bw() + 
+  coord_fixed() + 
+  guides(color = guide_legend(override.aes=list(size=4, alpha=1)))
+
+pc1vs2_regular <- 
+cbind(data.table(sem_comb@meta.data)[,.(cell_ID, batchid, x_slide_mm, y_slide_mm)]
+      ,data.table(sem_comb@reductions$pca@cell.embeddings))
+
+regular_pca_plt <- 
+ggplot(pc1vs2_regular, aes(x = PC_1, y = PC_2, color=batchid)) + geom_point(size=0.2, alpha=0.5, shape=16) + 
+  scale_color_manual(values = c("purple", "darkgreen")) + 
+  theme_bw() + 
+  coord_fixed() + 
+  guides(color = guide_legend(override.aes=list(size=4, alpha=1)))
+
+pc_comparison_plt <- 
+cowplot::plot_grid(harmony_plt + labs(title = "integrated using harmony")
+                   ,regular_pca_plt + labs(title = "un-integrated")) + 
+  theme(plot.background = element_rect(fill = 'white',color='white')) + 
+  cowplot::panel_border(remove=TRUE) 
+
+print(pc_comparison_plt)
```

Comment:
optional: if convenient, remake the plot so the purple batch doesn't cover the green batch.
Also, I feel like those dark colors aren't terribly high-contrast with each other.